strong-config/node
# Allow defining an optional base config that other configs can inherit from

### Description

Fixes #4 

Environment-specific configs (e.g. `development.yml`, `staging.yml`) can now inherit values from an optional base config (e.g. `base.yml`). This reduces a lot of duplication for config values that are the same across several environments.

**Caveat: You can not put secrets into the base config because it's not possible to determine which encryption key should be used for them because base configs are env-independent**

### How to test

1. Check out this branch locally
2. Check out the `./src/integration-tests/load-es6.ts` file and have a look at the second example 'LOAD WITH BASE CONFIG'
3. Check out the `./example-with-base-config/base.yml` and `./example-with-base-config/development.yml`
4. When loading the development.yml, we should also see the values from the base.yml in the final config
5. Run the integration test with `yarn test:core:load:es6`
6. Check that the final output on the console spits out a merged config object of base.yml + development.yml
